### PR TITLE
FIX: allow Python cache selection in actions

### PR DESCRIPTION
**Pull request by jorgepiloto**

### CODE REVIEW

File: _release-pypi/action.yml [hunk 1]
```diff
@@ -22,6 +22,10 @@ inputs:
     description: 'Desired Python version for using twine'
     required: true
     type: string
+  use-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
+    required: false
+    type: boolean
```
Comment (RobPasMue) [inline _release-pypi/action.yml:28]: You may need to add the default option.

Comment (RobPasMue) [reply _release-pypi/action.yml:28]: I would default it to false probably. But that's just my point of view 😄

File: build-library/action.yml [hunk 2]
```diff
@@ -11,6 +11,11 @@ inputs:
     default: '3.10'
     required: false
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline build-library/action.yml:15]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: build-wheelhouse/action.yml [hunk 3]
```diff
@@ -18,6 +18,11 @@ inputs:
     description: 'Desired Python version for build'
     required: true
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline build-wheelhouse/action.yml:22]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: code-style/action.yml [hunk 4]
```diff
@@ -7,6 +7,11 @@ inputs:
     default: '3.10'
     required: false
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline code-style/action.yml:11]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: doc-build/action.yml [hunk 5]
```diff
@@ -7,6 +7,11 @@ inputs:
     default: '3.10'
     required: false
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline doc-build/action.yml:11]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: doc-deploy-stable/action.yml [hunk 6]
```diff
@@ -17,6 +17,11 @@ inputs:
     default: '3.10'
     required: false
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline doc-deploy-stable/action.yml:21]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: release-pypi-public/action.yml [hunk 7]
```diff
@@ -19,6 +19,11 @@ inputs:
     required: false
     default: '3.10'
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline release-pypi-public/action.yml:23]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: release-pypi-test/action.yml [hunk 8]
```diff
@@ -19,6 +19,11 @@ inputs:
     required: false
     default: '3.10'
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline release-pypi-test/action.yml:23]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: tests-pytest/action.yml [hunk 9]
```diff
@@ -7,6 +7,11 @@ inputs:
     default: '3.10'
     required: false
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline tests-pytest/action.yml:11]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: _setup-python/action.yml [hunk 10]
```diff
@@ -6,14 +6,25 @@ inputs:
     description: 'Desired Python version'
     required: true
     type: string
+  use-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline _setup-python/action.yml:10]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: _release-pypi/action.yml [hunk 11]
```diff
@@ -22,6 +22,11 @@ inputs:
     description: 'Desired Python version for using twine'
     required: true
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline _release-pypi/action.yml:26]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: _release-pypi/action.yml (hunk 1 above)
Comment (jorgepiloto) [reply _release-pypi/action.yml]: I just imposed required parameters for "private" actions.

Actions using this one have a `use-python-cache = true` by default. This way we avoid passing one additional parameter and try to speed up our CI.

File: _setup-python/action.yml (hunk 10 above)
Comment (RobPasMue) [inline _setup-python/action.yml:10]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: build-ci-wheels/action.yml [hunk 12]
```diff
@@ -7,6 +7,11 @@ inputs:
     default: '3.10'
     required: false
     type: string
+  use-python-cache:
+    description: 'Whether to use Python chache or not. Requires a pyproject.toml file.'
```
Comment (RobPasMue) [inline build-ci-wheels/action.yml:11]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: build-library/action.yml (hunk 2 above)
Comment (RobPasMue) [inline build-library/action.yml:15]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: build-wheelhouse/action.yml (hunk 3 above)
Comment (RobPasMue) [inline build-wheelhouse/action.yml:22]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: code-style/action.yml (hunk 4 above)
Comment (RobPasMue) [inline code-style/action.yml:11]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: doc-build/action.yml (hunk 5 above)
Comment (RobPasMue) [inline doc-build/action.yml:11]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: doc-deploy-stable/action.yml (hunk 6 above)
Comment (RobPasMue) [inline doc-deploy-stable/action.yml:21]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: release-pypi-public/action.yml (hunk 7 above)
Comment (RobPasMue) [inline release-pypi-public/action.yml:23]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: release-pypi-test/action.yml (hunk 8 above)
Comment (RobPasMue) [inline release-pypi-test/action.yml:23]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

File: tests-pytest/action.yml (hunk 9 above)
Comment (RobPasMue) [inline tests-pytest/action.yml:11]:
```suggestion
    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
```

**Review by RobPasMue — APPROVED** [review]
I think you force-pushed and deleted my previous suggestions haha! LGTM, though I do not understand why it keeps failing....

**Comment by jorgepiloto** [comment]
Ops, sorry! I bet I had fetched upstream changes... Let me fix this.

**Pull request merged**